src/mono/wasm/runtime/hybrid-globalization.ts

```diff
@@ -135,24 +135,115 @@ export function mono_wasm_ends_with(exceptionMessage: Int32Ptr, culture: MonoStr
     }
 }
 
-function get_clean_string(strPtr: number, strLen: number) {
-    const str = string_decoder.decode(<any>strPtr, <any>(strPtr + 2 * strLen));
+function decode_to_clean_string(strPtr: number, strLen: number)
+{
+    const str = string_decoder.decode(<any>strPtr, <any>(strPtr + 2*strLen));
+    return clean_string(str);
+}
+
+function clean_string(str: string)
+{
     const nStr = str.normalize();
     return nStr.replace(/[\u200B-\u200D\uFEFF\0]/g, "");
 }
 
+export function mono_wasm_index_of(exceptionMessage: Int32Ptr, culture: MonoStringRef, needlePtr: number, needleLength: number, srcPtr: number, srcLength: number, options: number, fromBeginning: number): number{
+    const cultureRoot = mono_wasm_new_external_root<MonoString>(culture);
+    try {
+        const needle = string_decoder.decode(<any>needlePtr, <any>(needlePtr + 2*needleLength));
+        // no need to look for an empty string
+        if (clean_string(needle).length == 0)
```

Comment:
Could this move to C# side ? At least for simple case when there are no special chars.

Reply:
Ups, no - we cannot. If the source is empty then the repsonse depends on the target - if it's also empty after cleaning then we return 0 (or lastIndex) but if it's not empty after cleaning then we return -1. So we cannot make this decision in C#.